src/bootstrap/dist.rs

```diff
@@ -2297,6 +2297,48 @@ impl Step for RustDev {
     }
 }
 
+// Tarball intended for internal consumption to ease rustc/std development.
+//
+// Should not be considered stable by end users.
+#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+pub struct RustDevConfig {
+    pub target: TargetSelection,
+}
+
+impl Step for RustDevConfig {
+    type Output = Option<GeneratedTarball>;
+    const DEFAULT: bool = true;
+    const ONLY_HOSTS: bool = true;
+
+    fn should_run(run: ShouldRun<'_>) -> ShouldRun<'_> {
+        run.alias("rust-dev-config")
+    }
+
+    fn make_run(run: RunConfig<'_>) {
+        run.builder.ensure(RustDevConfig { target: run.target });
+    }
+
+    fn run(self, builder: &Builder<'_>) -> Option<GeneratedTarball> {
+        let target = self.target;
+
+        /* run only if llvm-config isn't used */
+        if let Some(config) = builder.config.target_config.get(&target) {
+            if let Some(ref _s) = config.llvm_config {
+                builder.info(&format!("Skipping RustDevConfig ({}): external LLVM", target));
+                return None;
+            }
+        }
```

Comment:
can you make this a default instead of a skip, so that we can give a hard error if someone runs `x dist rust-dev-config` when external llvm is set? see https://github.com/rust-lang/rust/pull/113640 for an example.

Reply:
Like this?
```rust
fn should_run(run: ShouldRun<'_>) -> ShouldRun<'_> {
        let config = run.builder.config.target_config.get(&<how to get target?>).map(|c| c.llvm_config);
        run.alias("rust-dev-config").default_condition(config.is_none())
    }
```
I'm not sure how to get the target, because `run.builder` contains a list of targets.